### ISSUE TITLE
[io] vtk-export for paraview

### ISSUE THREAD

**Issue by vacary** [siconos/siconos#59]
Wish list (again) 
- add velocity, position, kinetic energy attached to a cell in vtk export for paraview
- add location of contact points

**Comment by vacary** [comment]
ok, the preview export is working but it is impraticable for large results file (more than 100 cubes)

How to improve the efficiency of the export ?

**Comment by radarsat1** [comment]
I looked into this a bit but it seems very non-trivial without a fairly important reworking of VView.  Perhaps it could be based on vtkTemporalDataSet https://www.vtk.org/Wiki/VTK/Time_Support

**Comment by bremond** [comment]
Yes, a vtkTemporalDataSet would have really helped us, but ... it is no
longer used in recent versions of vtk:
https://www.vtk.org/Wiki/VTK/Remove_vtkTemporalDataSet

For me the solution lies now in programmable filters in Python (they
were not available in packaged vtk when I started to write vview):

https://blog.kitware.com/vtkpythonalgorithm-is-great/

https://blog.kitware.com/developing-hdf5-readers-using-vtkpythonalgorithm/

We could have this kind of reader, with time support. It would read
slices on demand from the time slider of vview or paraview.

This would need to have streaming time:

http://berkgeveci.github.io/2014/11/09/streaming-time/

with block base streaming in hdf5

http://berkgeveci.github.io/2015/01/19/block-streaming/

Everything seems to be here, so maybe it is not a huge work, probably
possible for 4.3 release (not 4.2 ...)

**Comment by bremond** [comment]
Here is adaptation of vview and vexport with a VTKPythonAlgorithmBase as the base reader.
The data is only read on demand at each time step inside the reader. Some slices are still needed to get times, and mu coefficients, but It is faster at startup on chute-* examples and profiling shows that time is mostly spent in the renderer now.
It provides an export of contact forces to paraview with a 2 blocks datasets: one for the data (which can be another multiblock or a unique polydata if with global-filter option) and one for contact points and associated vectors (normal, forces).  It is possible to see contact forces with the help of an extractBlock filter and a glyph filter. Domain info is not implemented.
https://github.com/bremond/siconos/tree/new-vview
![screenshot from 2018-12-10 17-33-34](https://user-images.githubusercontent.com/5980800/49747233-40184000-fca3-11e8-95fe-4005bbf21a56.png)

**Comment by bremond** [comment]
I made some enhancements with global-filter option (global-filter option is available with siconos_vview and siconos_vexport and it merges all objects in one big vtkPolyData, it is meant for the cases where the number of objects is huge). The attached arrays like velocity, displacement, and whatever we want to attach to each body, are usable now from paraview. They were usable only without global-filter option.
 I also added an instance data to mimick the coloring of the differents objects as for when --global-filter is not set (cf the attached image)
A few issues:
 - translation and displacement are the same arrays. which one should we keep ? I prefer translation as it is  shorter and is used every where in the code,
- velocity data is bugged, I do not know where it comes from,
- domain data for contact forces is still commented, do we have an example which used it so I can test ? (I think it is @radarsat1 who made this).

![screenshot from 2018-12-19 11-33-38](https://user-images.githubusercontent.com/5980800/50215415-3b2e4d00-0383-11e9-8406-36179a2983d6.png)
